RHIDP-1471: add GitHub auth provider content (#91)
redhat-developer/red-hat-developers-documentation-rhdh: getting-started/con-github-app-overview.adoc

= GitHub App​ overview

> GitHub Apps manage OAuth scope at the app installation level, rendering the scope parameter ineffective in the frontend. Nevertheless, it is crucial to include the necessary scope in open source plugins and document the required scopes for GitHub Apps in the plugin README.

Reviewer: Can we please add the location of the readme?
```suggestion
GitHub Apps manage OAuth scope at the application installation level, rendering the scope parameter ineffective in the front end. Nevertheless, you must include the necessary scope in open-source plugins and document the required scopes for GitHub Apps in the plugin README.
```

> NOTE: GitHub Apps are generally preferred to OAuth apps because they use fine-grained permissions, give more control over which repositories the app can access, and use short-lived tokens.

Reviewer: ```suggestion
NOTE: GitHub Apps are generally preferred to OAuth apps because they use fine-grained permissions, give more control over which repositories the application can access, and use short-lived tokens.
```
